src/Microsoft.Azure.WebJobs.Host/Config/ExtensionOptionsInfo.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Host.Config
+{
+    public class ExtensionOptionsInfo<TOptions> : IExtensionOptionsInfo
```

Comment:
As in my initial suggestion for this type, make it non-generic, and remove the interface. We can just register instances under service type ExtensionOptionsInfo, and the type is then easily extensible in the future.

Reply:
Hi @mathewc  
I understood your suggestion, however, I was not sure how we can register multiple `ExtensionOptionInfo` instances on the DI? For example.  The current registration using TryAddEnumerable to register multiple `ExtensionOptionsInfo<T>` instances with `IExtensionOptionsInfo`.  However, we can't register something like this `ServiceDescriptor.Singleton<ExtensionOptionInfo, ExtensionOptionInfo>(p => new ExtensionOptionInfo(builder.ExtensionInfo));`. Is there any other way to register multiple instances of the same class?
```
builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IExtensionOptionsInfo, ExtensionOptionsInfo<TOptions>>(p => new ExtensionOptionsInfo<TOptions>(builder.ExtensionInfo)));
```

Reply:
I see - yes, it seems we need an interface then. @fabiocav what are your thoughts on this? As part of the options configuration pipeline, we're registering instances of this type to enable us to later query all of them to discover all extensions/option Types registered. Is it strange to register "info" types like this? Usually types registered are services that provide behavior.

I guess another option would be for us to register a singleton IExtensionOptionsProvider (or some such name), and that service is responsible for returning the ExtensionOptionInfos.

Reply:
I talked with @fabiocav and change to the current design.